### PR TITLE
front: fix-bug: display rolling stock images

### LINKED ISSUE #3695: front: rolling stock images are not displayed anymore
### What happened?

Rolling stock images are not displayed anymore

### What did you expect to happen?

Rolling stock images should be displayed.

### How can we reproduce it (as minimally and precisely as possible)?

1. Open a scenario
2. Open the rolling stock selector
3. The images are not displayed

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Nixos v22.11
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

2b511b41

### PR DESCRIPTION
**Pull request by clarani**
closes #3695 

I tested my PR in local and in local with dev.

### CODE REVIEW

**Review by bloussou — DISMISSED** [review]
:rocket:

**Review by nicolaswurtz — CHANGES_REQUESTED** [review]
Use new marvelous RTK queries pleeeease :D

File: front/src/common/RollingStockSelector/RollingStock2Img.jsx [hunk 1]
```diff
@@ -18,9 +18,12 @@ function RollingStock2Img(props) {
...
-      });
+      const image = await get(
+        `/editoast/rolling_stock/${rollingStock.id}/livery/${defaultLivery.id}/`,
+        {
+          responseType: 'blob',
+        }
+      );
       if (image) setImageUrl(URL.createObjectURL(image));
     } catch (e) {
       /* empty */
```
Comment (nicolaswurtz) [inline front/src/common/RollingStockSelector/RollingStock2Img.jsx]: No empty catch error (at least put a `console.error`, thx !)

File: front/src/common/RollingStockSelector/RollingStock2Img.tsx [hunk 2]
```diff
@@ -1,44 +1,38 @@
...
+  };
 
   useEffect(() => {
-    setImageUrl(null);
+    setImageUrl('');
     getRollingStockImage();
-  }, [getRollingStockImage]);
+  }, [rollingStock]);
 
   return imageUrl ? <LazyLoadImage src={imageUrl} alt={rollingStock?.name} key={nextId()} /> : null;
```
Comment (nicolaswurtz) [inline front/src/common/RollingStockSelector/RollingStock2Img.tsx]: Please use something else than `nextId()`, why not create a unique id key with a prefix associated to rolling stock unique name ?

**Review by nicolaswurtz — APPROVED** [review]

**Pull request merged**